src/Installer/Microsoft.Dotnet.Installation/Internal/DotnetArchiveDownloader.cs

```diff
@@ -0,0 +1,259 @@
+
+    public DotnetArchiveDownloader()
+        : this(CreateDefaultHttpClient())
+    {
+    }
+
+    public DotnetArchiveDownloader(ReleaseManifest releaseManifest)
+        : this(CreateDefaultHttpClient())
+    {
+        _releaseManifest = releaseManifest ?? throw new ArgumentNullException(nameof(releaseManifest));
+    }
+
+    /// <summary>
+    /// Creates an HttpClient with enhanced proxy support for enterprise environments.
+    /// </summary>
+    private static HttpClient CreateDefaultHttpClient()
+    {
+        var handler = new HttpClientHandler()
+        {
+            UseProxy = true,
+            UseDefaultCredentials = true,
+            AllowAutoRedirect = true,
+            MaxAutomaticRedirections = 10,
+            AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate
+        };
+
+        var client = new HttpClient(handler)
+        {
+            Timeout = TimeSpan.FromMinutes(10)
+        };
+
+        // Set user-agent to identify dnup in telemetry
+        client.DefaultRequestHeaders.UserAgent.ParseAdd("dnup-dotnet-installer");
+
+        return client;
+    }
+
+    /// <summary>
+    /// Downloads the archive from the specified URL to the destination path with progress reporting.
+    /// </summary>
+    /// <param name="downloadUrl">The URL to download from</param>
+    /// <param name="destinationPath">The local path to save the downloaded file</param>
+    /// <param name="progress">Optional progress reporting</param>
+    /// <returns>True if download was successful, false otherwise</returns>
+    protected async Task<bool> DownloadArchiveAsync(string downloadUrl, string destinationPath, IProgress<DownloadProgress>? progress = null)
+    {
+        // Create temp file path in same directory for atomic move when complete
+        string tempPath = $"{destinationPath}.download";
+
+        for (int attempt = 1; attempt <= MaxRetryCount; attempt++)
+        {
+            try
+            {
+                // Ensure the directory exists
+                Directory.CreateDirectory(Path.GetDirectoryName(destinationPath)!);
+
+                // Try to get content length for progress reporting
+                long? totalBytes = await GetContentLengthAsync(downloadUrl);
+
+                // Make the actual download request
+                using var response = await _httpClient.GetAsync(downloadUrl, HttpCompletionOption.ResponseHeadersRead);
+                response.EnsureSuccessStatusCode();
+
+                // Get the total bytes if we didn't get it before
+                if (!totalBytes.HasValue && response.Content.Headers.ContentLength.HasValue)
+                {
+                    totalBytes = response.Content.Headers.ContentLength.Value;
+                }
+
+                using var contentStream = await response.Content.ReadAsStreamAsync();
+                using var fileStream = new FileStream(tempPath, FileMode.Create, FileAccess.Write, FileShare.None, 8192, true);
```

Comment:
Any specific reason for using 8192 as the file buffer size?

Reply:
8192 is a common maximum packet payload size https://datatracker.ietf.org/doc/html/rfc2734 however, I doubt that's the optimal choice.  This magic number should definitely be documented. In theory, we could make an issue describing that we should experiment with this, but I doubt it really matters much